<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="camera_id" default="camera" />
  <arg name="nn_model_path" default="$(find aw_platform)/config/model-050-0.8824.hdf5" />
  <arg name="image_transport" default="raw" />
  <arg name="image_raw_topic" default="$(arg camera_id)/image_raw "/>
  <arg name="roi_topic" default="/roi_signal" />
  <arg name="use_converted_map" default="false" />
  <arg name="change_state_threshold" default="2" />
  <arg name="score_threshold" default="0.6" />
  <arg name="use_peoria_hacks" default="false" />

  <!-- ROI extractor -->
  <node pkg="trafficlight_recognizer" type="region_tlr_tensorflow" name="region_tlr_tensorflow">
    <param name="image_transport" value="$(arg image_transport)" />
    <param name="image_raw_topic" value="$(arg image_raw_topic)" />
    <param name="roi_topic" value="$(arg roi_topic)" />
    <param name="use_converted_map" value="$(arg use_converted_map)" />
    <param name="change_state_threshold" value="$(arg change_state_threshold)" />
    <param name="score_threshold" value="$(arg score_threshold)" />
    <param name="use_peoria_hacks" value="$(arg use_peoria_hacks)" />
  </node>

  <!-- Classifier -->
  <node pkg="trafficlight_recognizer" type="tensorflow_tlr.py" name="tensorflow_tlr">
    <param name="nn_model_path" value="$(arg nn_model_path)" />
  </node>
</launch>
